src/ucp/proto/proto_multi.c

```diff
@@ -100,11 +100,11 @@ ucs_status_t ucp_proto_multi_init(const ucp_proto_multi_init_params_t *params,
         ucp_proto_common_lane_priv_init(&params->super, mpriv->reg_md_map, lane,
                                         &lpriv->super);
         max_frag = ucs_double_to_sizet(lanes_bandwidth[lane] / max_frag_ratio,
-                                       SIZE_MAX);
+                                       lanes_max_frag[lane]);
         ucs_assert(max_frag > 0);
 
         /* Due to floating-point accuracy, max_frag may be too large */
-        lpriv->max_frag = ucs_min(max_frag, lanes_max_frag[lane]);
+        lpriv->max_frag = max_frag;
```

Comment:
minor: maybe no need for `max_frag` now? can initialize `lpriv->max_frag` in place

Reply:
it looks better with code formatting.. and will be needed in next PR